.github/workflows/reuse-container-ci.yaml

```diff
@@ -0,0 +1,142 @@
+        required: false
+        type: string
+      runner_label:
+        description: 'Enter Validation Runner Label:'
+        default: test-runner
+        required: true
+        type: string
+  workflow_call:
+    inputs:
+      group_dir:
+        required: true
+        type: string
+jobs:
+####################################################################################################
+# Compose Build
+####################################################################################################
+  build-containers:
+    runs-on: ubuntu-latest # local registry label
+    outputs:
+      group: ${{ steps.build-group.outputs.container-group }}
+    steps:
+    - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+    - uses: docker/login-action@0d4c9c5ea7693da7b068278f7b52bda2a190a446 # v3.2.0
+      with:
+        registry: ${{ secrets.REGISTRY }}
+        username: ${{ secrets.REGISTRY_USER }}
+        password: ${{ secrets.REGISTRY_TOKEN }}
+      if: ${{ !inputs.no_build }}
+    - name: Build Container Group
+      if: ${{ !inputs.no_build }}
+      id: build-group
+      uses: ./.github/workflows/composite/docker-build
+      with:
+        group_dir: ${{ inputs.group_dir }}
+        env_overrides: ${{ inputs.env_overrides || env.env_overrides || '' }}
+        registry: ${{ secrets.REGISTRY }}
+####################################################################################################
+# Trivy Scan
+####################################################################################################
+  setup-scan:
+    needs: [build-containers]
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.scan-matrix.outputs.matrix }}
+    steps:
+    - name: Harden Runner
+      uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6 # v2.8.1
+      with:
+        egress-policy: audit
+    - uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+      with:
+        path: matrix
+    - name: Set Matrix
+      id: scan-matrix
+      run: echo "matrix=$(cat matrix/*-${{ needs.build-containers.outputs.group }}/*.txt | jq -R '.' | jq -sc '. | unique')" >> $GITHUB_OUTPUT
+  scan-containers:
+    needs: [setup-scan]
+    if: ${{ !inputs.no_build && github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest # local registry label
+    env:
+      GROUP_DIR: ${{ inputs.group_dir }}
+    strategy:
+      matrix:
+        container: ${{ fromJSON(needs.setup-scan.outputs.matrix) }}
+      fail-fast: false
+    steps:
+    - name: Harden Runner
+      uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6 # v2.8.1
+      with:
+        egress-policy: audit
+    - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+    - uses: docker/login-action@0d4c9c5ea7693da7b068278f7b52bda2a190a446 # v3.2.0
+      with:
+        registry: ${{ secrets.REGISTRY }}
```

Comment:
How to use a common registry on IDC machine? Now we have 2 local registry separately for xeon and gaudi cluster.

Reply:
If you can use something like harbor that can be accessed by both clusters or a cloud registry that would be ideal.

If your runners have pre-authenticated access to these local registries, then we could remove the login commands/secrets and use environment variables that are present in the runner environment.